apis/v1alpha1/backendpolicy_types.go

```diff
@@ -0,0 +1,173 @@
+/*
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+
+// BackendPolicy defines policies associated with backends. For the purpose of
+// this API, a backend is defined as any resource that a route can forward
+// traffic to. A common example of a backend is a Service. Configuration that is
+// implementation specific may be represented with similar implementation
+// specific custom resources.
+type BackendPolicy struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   BackendPolicySpec   `json:"spec,omitempty"`
+	Status BackendPolicyStatus `json:"status,omitempty"`
+}
+
+// +kubebuilder:object:root=true
+
+// BackendPolicyList contains a list of BackendPolicy
+type BackendPolicyList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []BackendPolicy `json:"items"`
+}
+
+// BackendPolicySpec defines desired policy for a backend.
+type BackendPolicySpec struct {
+	// BackendRefs define which backends this policy should be applied to. If
+	// more than one BackendPolicy targets the same backend, precedence will be
+	// given to the oldest BackendPolicy.
+	//
+	// Support: Core
+	//
+	// +optional
+	BackendRefs []BackendRef `json:"backendRefs"`
```

Comment:
Just to clarify, an BackendPolicy object only apply to backends in the same namespace, correct?

Reply:
A very important question. My understanding here is yes, namespace restrictions apply. Is that the case, Rob?

Reply:
> // BackendRef identifies an API object within a known namespace that defaults
// group to core and resource to services if unspecified

This is what we are looking for I think, @yiyangy.

Reply:
+1 that a BackendPolicy object should only apply to backends in the same namespace.

Reply:
@robscott should `BackendRefs` be required instead of optional? What is the purpose of `BackendPolicySpec` without specifying `BackendRefs`?

Reply:
Updated to make this required and clarify that this is for same-namespace backends only - thanks for the feedback!